velox/functions/prestosql/tests/TimestampWithTimeZoneCastTest.cpp

```diff
@@ -64,6 +65,82 @@ TEST_F(TimestampWithTimeZoneCastTest, fromTimestamp) {
   testCast(tsVector, expected);
 }
 
+TEST_F(TimestampWithTimeZoneCastTest, fromVarchar) {
+  const auto stringVector = makeNullableFlatVector<StringView>(
+      {std::nullopt,
+       "2012-10-31 01:00:47 America/Denver",
+       "2012-10-31 01:00:47 -06:00",
+       "1994-05-06 15:49 Europe/Vienna",
+       "1979-02-24 08:33:31 Pacific/Chatham",
+       "1979-02-24 08:33:31 +13:45"});
```

Comment:
can you also add a test case with a string without a timezone, to ensure it falls back to the session timezone?

Reply:
I think I see what you mean now - currently it wouldn't default to the session timezone, I didn't realize we wanted to support casting of a string without a timezone. 

If so, I see the potential use for the `adjustTimestampToTimezone` flag - it could dictate behavior upon lack of a timezone in the string, whether we fall back to session timezone or fail the cast. Maybe it would be worthwhile to see how Presto Java is handling this scenario?

Reply:
Yes. Please validate how Presto java behaves in these cases and follow the same semantic.

Reply:
@pedroerp seems that Java is indeed falling back to session timezone in the case of an input string containing no timezone - https://github.com/prestodb/presto/blob/6af662d92e5b0fa5b651db67dc00a747edb64bdf/presto-main/src/main/java/com/facebook/presto/util/DateTimeUtils.java#L204.

Replicated the same behavior here and added a test case to validate.